File: row-change-event-channel/src/test/java/nl/haploid/octowight/service/EventChannelServiceIT.java
package nl.haploid.octowight.service;

import nl.haploid.octowight.AbstractIT;
import nl.haploid.octowight.AtomChangeEvent;
import nl.haploid.octowight.TestData;
import org.apache.kafka.clients.producer.RecordMetadata;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.Future;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class EventChannelServiceIT extends AbstractIT {

	@Autowired
	private EventChannelService service;

	@Test
	public void testSendEvent() throws Exception {
		final AtomChangeEvent event = TestData.atomChangeEvent();
		final Future<RecordMetadata> future = service.sendEvent(event);
		assertNotNull(future.get());
	}

	@Test
	public void testSendEvents() throws Exception {
		final AtomChangeEvent event1 = TestData.atomChangeEvent();
		final AtomChangeEvent event2 = TestData.atomChangeEvent();
		final List<RecordMetadata> results = service.sendEvents(Arrays.asList(event1, event2));
		final int expectedCount = 2;
		final int actualCount = results.size();
		assertEquals(expectedCount, actualCount);
	}
}

File: event-emitter/src/test/scala/nl/haploid/octowight/service/EventChannelServiceIT.scala
package nl.haploid.octowight.service

import java.util.concurrent.TimeUnit

import nl.haploid.octowight.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class EventChannelServiceIT extends AbstractIT {
  @Autowired private[this] val eventChannelService: EventChannelService = null

  "Channel" should "send an event" in {
    val event = TestData.atomChangeEvent
    val future = eventChannelService.sendEvent(event)
    Option(future.get(5, TimeUnit.SECONDS)) should not be None
  }

  "Channel" should "send multiple events" in {
    val results = eventChannelService.sendEvents(List(TestData.atomChangeEvent, TestData.atomChangeEvent))
    results should have size 2
  }
}
